### PR TITLE
[done] PMD #3

### PR DESCRIPTION
**Pull request by TheRealMarnes**
Last of the "a million superficial fixes" PMD PRs :)

I expect these last rules may include some disagreeable ones, though I did my best to avoid them. What remains will likely just stay excluded because they're controversial or not useful, or need major work to fix. There's some use of `@SuppressWarnings` because there honestly were a few legit cases where the rules needed to be relaxed a little, or I'd have to break the API and stuff.

Last commit in the series is one you might very understandably object to (performance vs clarity/brevity), and accounts for a decent chunk of the very reasonable lines-changed count.

### CODE REVIEW

File: core/src/main/java/org/jdbi/v3/core/statement/AbstractBaseStatement.java [hunk 1]
```diff
@@ -23,21 +23,21 @@
 import org.jdbi.v3.core.config.Configurable;
 import org.jdbi.v3.core.generic.GenericTypes;
 
-abstract class BaseStatement<This> implements Closeable, Configurable<This> {
+abstract class AbstractBaseStatement<This> implements Closeable, Configurable<This> {
```
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/core/statement/AbstractBaseStatement.java]: I don't love the change adding `Abstract` to all the class names.  Is there a particular reason to do this?

File: core/src/main/java/org/jdbi/v3/core/statement/Binding.java [hunk 2]
```diff
@@ -89,42 +89,19 @@ public void addNamedArgumentFinder(NamedArgumentFinder args) {
...
-        if (wrote) {
-            wrote = false;
-            b.deleteCharAt(b.length() - 1);
-        }
-        b.append("}");
+        String named = this.named.entrySet().stream()
+            .map(x -> x.getKey() + ':' + x.getValue())
+            .collect(Collectors.joining(","));
 
-        b.append(", finder:[");
```
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/core/statement/Binding.java]: Nice

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/config/internal/UseSqlParserImpl.java [hunk 3]
```diff
@@ -26,43 +31,37 @@
...
-                return classArg.newInstance(sqlObjectType);
-            } catch (NoSuchMethodException e1) {
-                if (m != null) {
-                    Constructor<? extends SqlParser> constructor = value.getConstructor(Class.class,
-                                                                                            Method.class);
-                    return constructor.newInstance(sqlObjectType, m);
-                }
-                throw new IllegalStateException("Unable to instantiate, no viable constructor " + value.getName());
+                Class[] classes = Arrays.stream(nonNullArgs).map(Object::getClass).toArray(Class[]::new);
+                return c.getConstructor(classes).newInstance(nonNullArgs);
```
Comment (stevenschlansker) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/config/internal/UseSqlParserImpl.java]: If we're going to change this, we should change it to use the new `MethodHandles` style reflection.

File: core/src/main/java/org/jdbi/v3/core/statement/AbstractBaseStatement.java (hunk 1 above)
Comment (TheRealMarnes) [reply core/src/main/java/org/jdbi/v3/core/statement/AbstractBaseStatement.java]: Style convention. I won't argue about it if you don't like it.

**Comment by TheRealMarnes** [comment]
Naming reverted. I don't know what MethodHandles is, I'll have a look at that when I get the time

**Comment by stevenschlansker** [comment]
Take a look, we can split it into a separate PR if you like, that way the rest of this could get merged.

**Comment by TheRealMarnes** [comment]
Can we just merge this one to get it out of the way asap and I'll do the reflection update separately?

**Pull request merged**